src/openrct2-ui/windows/RideList.cpp

```diff
@@ -172,872 +143,795 @@ static constexpr const rct_string_id page_names[] = {
             | (1ULL << WIDX_TAB_3) | (1ULL << WIDX_CLOSE_LIGHT) | (1ULL << WIDX_OPEN_LIGHT);
         if (network_get_mode() != NETWORK_MODE_CLIENT)
         {
-            window->enabled_widgets |= (1ULL << WIDX_QUICK_DEMOLISH);
+            this->enabled_widgets |= (1ULL << WIDX_QUICK_DEMOLISH);
         }
-        WindowInitScrollWidgets(window);
-        window->page = PAGE_RIDES;
-        window->selected_list_item = -1;
-        window->frame_no = 0;
-        window->min_width = 340;
-        window->min_height = 240;
-        window->max_width = 400;
-        window->max_height = 700;
-        window_ride_list_refresh_list(window);
+        WindowInitScrollWidgets(this);
+        this->page = PAGE_RIDES;
+        this->selected_list_item = -1;
+        this->frame_no = 0;
+        this->min_width = 340;
+        this->min_height = 240;
+        this->max_width = 400;
+        this->max_height = 700;
+        RefreshList();
+
+        this->list_information_type = 0;
+
+        _windowRideListInformationType = INFORMATION_TYPE_STATUS;
+        _quickDemolishMode = false;
     }
 
-    window->list_information_type = 0;
-    _quickDemolishMode = false;
-
-    return window;
-}
-
-/**
- *
- *  rct2: 0x006B3511
- */
-static void window_ride_list_mouseup(rct_window* w, rct_widgetindex widgetIndex)
-{
-    switch (widgetIndex)
+    /**
+     *
+     *  rct2: 0x006B38A7
+     */
+    void OnResize() override
     {
-        case WIDX_CLOSE:
-            window_close(w);
-            break;
-        case WIDX_SORT:
-            w->list_information_type = _window_ride_list_information_type;
-            w->selected_list_item = -1;
-            window_ride_list_refresh_list(w);
-            break;
-        case WIDX_TAB_1:
-        case WIDX_TAB_2:
-        case WIDX_TAB_3:
-            if (w->page != widgetIndex - WIDX_TAB_1)
-            {
-                w->page = widgetIndex - WIDX_TAB_1;
-                w->frame_no = 0;
-                w->selected_list_item = -1;
-                if (w->page != PAGE_RIDES && _window_ride_list_information_type > INFORMATION_TYPE_RUNNING_COST)
-                {
-                    _window_ride_list_information_type = INFORMATION_TYPE_STATUS;
-                }
-                window_ride_list_refresh_list(w);
-            }
-            break;
-        case WIDX_CLOSE_LIGHT:
-            window_ride_list_close_all(w);
-            break;
-        case WIDX_OPEN_LIGHT:
-            window_ride_list_open_all(w);
-            break;
-        case WIDX_QUICK_DEMOLISH:
-            if (network_get_mode() != NETWORK_MODE_CLIENT)
-            {
-                _quickDemolishMode = !_quickDemolishMode;
-            }
-            else
-            {
-                _quickDemolishMode = false;
-            }
-            w->Invalidate();
-            break;
-    }
-}
+        this->min_width = 340;
+        this->min_height = 124;
```

Comment:
Why are they being constantly set here? It should be enough to do that once in the constructor.